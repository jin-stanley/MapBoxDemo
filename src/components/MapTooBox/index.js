/***
 * Using for FilterPanel and LayersPanel
 */

import React from "react";
import { makeStyles } from "@material-ui/core/styles";

import { ReactComponent as LayersLogo } from "../../img/svg/layers.svg";
import { ReactComponent as FiltersLogo } from "../../img/svg/levels.svg";

const MapTooBox = (props) => {
  const classes = useStyles();
  return (
    <React.Fragment>
      <div
        className={`${classes.filters} ${classes.settingWrapper}`}
        onClick={props.toggleFiltersPannel}
      >
        <FiltersLogo className={classes.logo} />
        <div>Filters</div>
      </div>
      <div
        className={`${classes.layers} ${classes.settingWrapper}`}
        onClick={props.toggleLayersPannel}
      >
        <LayersLogo className={classes.logo} />
        <div>Layers</div>
      </div>
    </React.Fragment>
  );
};

const useStyles = makeStyles({
  settingWrapper: {
    zIndex: 2,
    width: "64px",
    height: "64px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#fff",
    border: "1px solid #bbb",
    borderRadius: "8px",
    cursor: "pointer",

    "&:hover": {
      border: "1px solid #aaa",
      backgroundColor: "#1F4B5C",
      color: "#fff",
      "& svg": {
        fill: "#fff",
      },
    },
  },
  layers: {
    position: "fixed",
    bottom: "30px",
    right: "30px",
    display: "flex",
    flexDirection: "column",
  },
  filters: {
    position: "fixed",
    bottom: "30px",
    right: "120px",
    display: "flex",
    flexDirection: "column",
  },
  logo: {
    width: "24px",
    height: "24px",
    cursor: "pointer",
  },
});

export default MapTooBox;
