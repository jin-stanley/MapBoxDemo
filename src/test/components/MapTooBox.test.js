import React from "react";
import renderer from "react-test-renderer";

import MapTooBox from "../../components/MapTooBox";

it("MapTooBox can render success", () => {
  const tree = renderer.create(<MapTooBox />).toJSON();
  expect(tree).toMatchSnapshot();
});
