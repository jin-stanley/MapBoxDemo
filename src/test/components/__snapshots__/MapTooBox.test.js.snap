// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapTooBox can render success 1`] = `
Array [
  <div
    className="makeStyles-filters-3 makeStyles-settingWrapper-1"
  >
    <svg
      className="makeStyles-logo-4"
    >
      levels.svg
    </svg>
    <div>
      Filters
    </div>
  </div>,
  <div
    className="makeStyles-layers-2 makeStyles-settingWrapper-1"
  >
    <svg
      className="makeStyles-logo-4"
    >
      layers.svg
    </svg>
    <div>
      Layers
    </div>
  </div>,
]
`;
